### ISSUE TITLE
Registry allows multiple accessURL per publisher, but GN doesn't

### ISSUE THREAD

**Issue by robredpath** [OpenDataServices/grantnav#418]
In the 360 Registry, we consider accessURL to be a property of a distribution (see http://threesixtygiving.github.io/getdata/ ). However, GN assumes that there will only be one accessURL per publisher, and as a result uses the first accessURL in the data for publisher pages and on the datasets page (see https://github.com/OpenDataServices/grantnav/search?utf8=%E2%9C%93&q=accessurl&type= ) 

We should look at solutions to this! One might be to treat accessURL as a list in these contexts - a bit of design work would be necessary to ensure that it stays clear that GN just displays data from other sources and to make the conversion to a table work nicely, but I don't think this would be a huge deal. 

An alternative would be to just remove the accessURL from use in GrantNav - perhaps pointing people to the 360 Registry instead?

**Comment by KDuerden** [comment]
Thanks Rob. Just to note, current Registry on Wordpress page doesn't include access link: http://www.threesixtygiving.org/data/data-registry/
The JSON feed of the data does.

**Comment by morchickit** [comment]
I think removing it from GrantNav is probably the better option. We should look at it as a whole strategy.

**Comment by KDuerden** [comment]
It is a useful to have a link of that kind in GrantNav - as it guides users to the further information about their data.

**Comment by robredpath** [comment]
Part of the reason it's in there is to answer the question of "Who can I tell if there's a problem with the data?" - the person/team/department linked from the accessURL is more likely to be able to assist than having the user Google for the organisation name and contact a front desk.

**Comment by morchickit** [comment]
What's the status on this now?

**Comment by stevieflow** [comment]
I think.,,,, it's sorted, but would be helpful if @KDuerden to clarify

Take a look at https://grantnav.threesixtygiving.org/publisher/360G-cheshirecf

**Comment by KDuerden** [comment]
Not sorted, GN just picks one of the hosting pages to display: 
https://grantnav.threesixtygiving.org/publisher/360G-SuffolkCF

These are the 3 in use by this publisher.
https://www.suffolkcf.org.uk/grants-awarded-2017-18/
https://www.suffolkcf.org.uk/grants-awarded-2018-19/
https://www.suffolkcf.org.uk/suffolk-coronavirus-community-fund/

The Data Registry now does include a link to the hosting page for each file.

**Comment by stevieflow** [comment]
Thanks @KDuerden 

@robredpath given the progress made elsewhere on this, is this now feasible to fix in GN?

But also - @morchickit given the discussion in #29 this might also be another reason to deivert  back to the Registry (or other 360 asset)..?

**Comment by robredpath** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [GrantNav - show all accessURLs for a publisher](https://trello.com/c/TG6sdHcu/69-grantnav-show-all-accessurls-for-a-publisher)

**Comment by robredpath** [comment]
The recent work doesn't actually help us here - it's the same amount of work as it was back in 2017! This is a relatively straightforward fix that has just never worked its way up the list of priorities enough for us to do anything about. It's in the order of 'hours' of work to do. 

I've added it to Trello, and we can discuss priority against everything else we're up to the next time we look at the board, if we think it's important!

**Comment by robredpath** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Improve GrantNav static pages](https://trello.com/c/SGatdlHy/190-improve-grantnav-static-pages)

**Comment by mariongalley** [comment]
> Part of the reason it's in there is to answer the question of "Who can I tell if there's a problem with the data?" - the person/team/department linked from the accessURL is more likely to be able to assist than having the user Google for the organisation name and contact a front desk.

If we introduce a clear process for alerting publishers to a problem with their data, does that eliminate the user need for this field?

**Comment by robredpath** [comment]
> If we introduce a clear process for alerting publishers to a problem with their data, does that eliminate the user need for this field?

Hmmm. Maybe. 

I think it's important that if someone gets hold of a spreadsheet of grants data that uses the standard completely externally from 360Giving that they have the information that they need to understand what they have - that's a large part of the value of the Meta tab. 

The general principle of GrantNav is to be a mirror to the data, so if there's a good way to contact the publisher, it should be in the data and then reflected in GrantNav, potentially with suitable enhancements. If there's a process then ideally I think it would be visible in the data and GN would just provide a nice front-end for it. But that depends on what the process is, really!

**Comment by mariongalley** [comment]
Are there any cases where there is >1 access URL per funder @KDuerden ?

**Comment by KDuerden** [comment]
@mariongalley yes 32 publishers have two different Access URLs, a further 18 have three or more. There's opportunity for some cleaning to reduce this but not in all cases.
